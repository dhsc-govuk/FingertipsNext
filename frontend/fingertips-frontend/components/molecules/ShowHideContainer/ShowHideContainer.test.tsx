import { render } from '@testing-library/react';
import { ShowHideContainer } from '.';

describe('ShowHideContainer', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should not show the side bar when showSideBarWhenOpen is false', () => {
    /**
     * In the snapshot you will still the following style
     * .c1 div {
     *   border-left: none;
     * }
     * This indicates the sidebar was not rendered
     */

    const container = render(
      <ShowHideContainer summary="Some summary" showSideBarWhenOpen={false}>
        <div data-testid="inner-component">Some child element</div>
      </ShowHideContainer>
    );

    expect(container.asFragment()).toMatchSnapshot();
  });

  it('should not show the side bar when showSideBarWhenOpen is true', () => {
    /**
     * In the snapshot you will still the following style
     * .c1 div {
     *   border-left: 5px #A1A2A3 solid;
     *   padding-left: 1em;
     * }
     * This indicates the sidebar was rendered
     */

    const container = render(
      <ShowHideContainer summary="Some summary" showSideBarWhenOpen={true}>
        <div data-testid="inner-component">Some child element</div>
      </ShowHideContainer>
    );

    expect(container.asFragment()).toMatchSnapshot();
  });
});
