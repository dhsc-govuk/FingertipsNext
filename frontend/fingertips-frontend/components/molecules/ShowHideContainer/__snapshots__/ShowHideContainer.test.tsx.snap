// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowHideContainer should not show the side bar when showSideBarWhenOpen is false 1`] = `
<DocumentFragment>
  .c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  margin-bottom: 20px;
  display: block;
}

.c3 {
  display: inline-block;
  position: relative;
  margin-bottom: 5px;
  padding-left: 25px;
  color: #1d70b8;
  cursor: pointer;
}

.c3:hover {
  color: #2b8cc4;
}

.c3:focus {
  outline: 4px solid #ffdd00;
  outline-offset: -1px;
  color: #0b0c0c;
  background: #ffdd00;
}

.c3::-webkit-details-marker {
  display: none;
}

.c3:before {
  content: "";
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  margin: auto;
  display: block;
  width: 0;
  height: 0;
  border-style: solid;
  border-color: transparent;
  -webkit-clip-path: polygon(0% 0%,100% 50%,0% 100%);
  clip-path: polygon(0% 0%,100% 50%,0% 100%);
  border-width: 7px 0 7px 12.124px;
  border-left-color: inherit;
}

[open] > .c2:before {
  display: block;
  width: 0;
  height: 0;
  border-style: solid;
  border-color: transparent;
  -webkit-clip-path: polygon(0% 0%,50% 100%,100% 0%);
  clip-path: polygon(0% 0%,50% 100%,100% 0%);
  border-width: 12.124px 7px 0 7px;
  border-top-color: inherit;
}

.c4 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5 {
  padding: 15px;
  padding-left: 20px;
  border-left: 5px solid #bfc1c3;
}

.c5 p {
  margin-top: 0;
  margin-bottom: 20px;
}

.c5 > :last-child,
.c5 p:last-child {
  margin-bottom: 0;
}

.c1 div {
  padding: 0.5em 0em;
}

.c1 summary {
  color: #000000;
}

.c1 summary:hover {
  color: #000000;
}

.c1 div {
  border-left: none;
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c0 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 30px;
  }
}

<details
    class="c0 c1"
    open=""
  >
    <summary
      class="c2 c3"
    >
      <span
        class="c4"
      >
        Some summary
      </span>
    </summary>
    <div
      class="c5"
    >
      <div
        data-testid="inner-component"
      >
        Some child element
      </div>
    </div>
  </details>
</DocumentFragment>
`;

exports[`ShowHideContainer should not show the side bar when showSideBarWhenOpen is true 1`] = `
<DocumentFragment>
  .c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  margin-bottom: 20px;
  display: block;
}

.c3 {
  display: inline-block;
  position: relative;
  margin-bottom: 5px;
  padding-left: 25px;
  color: #1d70b8;
  cursor: pointer;
}

.c3:hover {
  color: #2b8cc4;
}

.c3:focus {
  outline: 4px solid #ffdd00;
  outline-offset: -1px;
  color: #0b0c0c;
  background: #ffdd00;
}

.c3::-webkit-details-marker {
  display: none;
}

.c3:before {
  content: "";
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  margin: auto;
  display: block;
  width: 0;
  height: 0;
  border-style: solid;
  border-color: transparent;
  -webkit-clip-path: polygon(0% 0%,100% 50%,0% 100%);
  clip-path: polygon(0% 0%,100% 50%,0% 100%);
  border-width: 7px 0 7px 12.124px;
  border-left-color: inherit;
}

[open] > .c2:before {
  display: block;
  width: 0;
  height: 0;
  border-style: solid;
  border-color: transparent;
  -webkit-clip-path: polygon(0% 0%,50% 100%,100% 0%);
  clip-path: polygon(0% 0%,50% 100%,100% 0%);
  border-width: 12.124px 7px 0 7px;
  border-top-color: inherit;
}

.c4 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5 {
  padding: 15px;
  padding-left: 20px;
  border-left: 5px solid #bfc1c3;
}

.c5 p {
  margin-top: 0;
  margin-bottom: 20px;
}

.c5 > :last-child,
.c5 p:last-child {
  margin-bottom: 0;
}

.c1 div {
  padding: 0.5em 0em;
}

.c1 summary {
  color: #000000;
}

.c1 summary:hover {
  color: #000000;
}

.c1 div {
  border-left: 5px #A1A2A3 solid;
  padding-left: 1em;
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c0 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 30px;
  }
}

<details
    class="c0 c1"
    open=""
  >
    <summary
      class="c2 c3"
    >
      <span
        class="c4"
      >
        Some summary
      </span>
    </summary>
    <div
      class="c5"
    >
      <div
        data-testid="inner-component"
      >
        Some child element
      </div>
    </div>
  </details>
</DocumentFragment>
`;
